lib/connection.js

```diff
@@ -963,19 +980,31 @@ Connection.prototype._close = function(force, callback) {
         if (err) {
           return callback(err);
         }
+        if (closeAndClean && _this.base.connections.indexOf(conn) != -1) {
+          console.log("BASE CONNECTIONS COUNT BEFORE 2---------> ", _this.base.connections.length)
```

Comment:
Same as above as well.

Reply:
@AbdelrahmanHafez Yes, I am testing out these changes, will remove them once testing is done before I mark this ready to merge.